### PR TITLE
feat(orchestrator): add fine-grained RBAC

### PR DESCRIPTION
**Pull request by mareklibra**
Fixes: [FLPATH-1562](https://issues.redhat.com//browse/FLPATH-1562)

The admin can limit access to specific workflows and instances by workflow ID.

TODO:
- [ ] Explore possibility to store user info along workflow instance

### CODE REVIEW

File: plugins/orchestrator/src/hooks/usePermissionArray.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+/**
+ * Like usePermission() from '@backstage/plugin-permission-react' but for multiple permissions at once.
+ *
+ * @param permissions
+ * @returns Object of the AsyncPermissionResult type which is ALLOWED if at least one of the permission is allowed.
+ */
+export const usePermissionArray = (
+  permissions: InputType[],
+): AsyncPermissionResult => {
+  const permissionApi = useApi(permissionApiRef);
```
Comment (mareklibra) [inline plugins/orchestrator/src/hooks/usePermissionArray.ts:25]: TODO: I will reuse the `usePermissionArrayBatch()` for that

**Comment by batzionb** [comment]
@mareklibra 
Does this use conditions?
https://backstage.io/docs/reference/plugin-permission-common.permissioncondition/#:~:text=A%20condition%20returned%20with%20a,claims%20from%20a%20identity%20token.

**Comment by mareklibra** [comment]
@batzionb This is still a draft.
I tried conditional permissions as a first option but was not able to get it working, this is a second attempt to have something working. But I keep trying.

**Comment by batzionb** [comment]
@mareklibra 
please reopen the PR on https://github.com/redhat-developer/rhdh-plugins

**Comment by mareklibra** [comment]
This PR has been moved to https://github.com/redhat-developer/rhdh-plugins/pull/82

**Pull request closed without merging**